common/components/AddressBookTable.tsx

```diff
@@ -0,0 +1,178 @@
+  temporaryLabel: string;
+  temporaryAddress: string;
+}
+
+class AddressBookTable extends React.Component<Props, State> {
+  public state: State = {
+    editingRow: null,
+    temporaryLabel: '',
+    temporaryAddress: ''
+  };
+
+  private addressInput: HTMLInputElement | null = null;
+
+  private labelInput: HTMLInputElement | null = null;
+
+  public render() {
+    const { rows } = this.props;
+    const { temporaryLabel, temporaryAddress } = this.state;
+
+    return (
+      <table className="AddressBookTable table" onKeyDown={this.handleKeyDown}>
+        <thead>
+          <tr>
+            <th scope="col">{translate('ADDRESS')}</th>
+            <th scope="col">{translate('LABEL')}</th>
+            <th scope="col">{translate('ACTIONS')}</th>
+          </tr>
+        </thead>
+        <tbody>
+          <tr>
+            <td>
+              <div className="AddressBookTable-cell">
+                <Input
+                  placeholder="New address"
+                  value={temporaryAddress}
+                  onChange={this.setTemporaryAddress}
+                  setInnerRef={this.setAddressInputRef}
+                />
+                <Identicon address={temporaryAddress} />
+              </div>
+            </td>
+            <td>
+              <div className="AddressBookTable-cell">
+                <Input
+                  placeholder="New label"
+                  value={temporaryLabel}
+                  onChange={this.setTemporaryLabel}
+                  setInnerRef={this.setLabelInputRef}
+                />
+              </div>
+            </td>
+            <td>
+              <div className="AddressBookTable-cell">
+                <button className="btn btn-sm btn-default" onClick={this.handleAddEntry}>
+                  <i className="fa fa-plus" />
+                </button>
+              </div>
+            </td>
+          </tr>
+          {rows.map(this.makeLabelRow)}
+        </tbody>
+      </table>
+    );
+  }
+
+  private handleSave = (addressToLabel: AddressLabelPair) => {
+    const { label, address } = addressToLabel;
+
+    this.props.addLabelForAddress({ label, address });
+
+    this.setEditingRow(null);
+  };
+
+  private handleAddEntry = () => {
+    const { temporaryLabel: label, temporaryAddress: address } = this.state;
+
+    if (!isValidETHAddress(address) && this.addressInput) {
+      return this.addressInput.focus();
+    }
+
+    if (!label && this.labelInput) {
+      return this.labelInput.focus();
+    }
+
+    if (label && isValidETHAddress(address)) {
+      this.handleSave({ label, address });
+      this.clearTemporaryFields();
+    }
+  };
+
+  private handleKeyDown = (e: React.KeyboardEvent<HTMLTableElement>) => {
+    if (e.keyCode === KeyCodes.ENTER) {
+      this.handleAddEntry();
+    }
+  };
+
+  private setEditingRow = (editingRow: number | null) => this.setState({ editingRow });
+
+  private removeEntry = (address: string) => this.props.removeLabelForAddress(address);
```

Comment:
The two above methods dont really seem needed, can just call the associated functionality directly

Reply:
I'll agree on `removeEntry`, but I like keeping `setState` calls in their own method as a personal quirk. I don't think it causes harm, and it improves readability.

:+1: on `removeEntry`